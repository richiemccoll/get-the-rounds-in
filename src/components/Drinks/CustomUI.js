import styled from 'styled-components';

export const Wrapper = styled.ul`
  padding: 0;
  margin: 0;
  list-style: none;
  display: flex;
  -webkit-flex-flow: row wrap;
  justify-content: center;

  li {
    margin-bottom: 1em;
  }

  @media (min-width: 900px) {
    justify-content: space-between;
  }
`;

export const Image = styled.img`
  height: 200px;
  width: 100%;
`;