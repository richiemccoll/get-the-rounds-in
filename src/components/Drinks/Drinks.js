import React from "react";
import { Card, Image, Button } from "semantic-ui-react";

class Drinks extends React.Component {

  constructor(props) {
    super(props);
    this.addDrink = this.addDrink.bind(this);
  }

  addDrink(drink) {
    this.props.addDrink(drink);
  }

  render() {
    const drinks = this.props.drinks.map((drink, index) => {
      return (
        <Drink
          key={index}
          name={drink[0].name}
          image={drink[0].image}
          price={drink[0].price}
          addDrink={this.addDrink}
        />
      );
    });

    return (
      <div>
        {drinks}
      </div>
    );
  }
}

const Drink = ({ name, image, price, addDrink }) => {
  return (
    <Card>
      <Image src={image} size="medium" height="175"/>
      <Card.Content>
        <Card.Header>
          {name}
        </Card.Header>
        <Card.Description>
          Â£{price.toFixed(2)}
        </Card.Description>
        <Card.Content extra>
          <div className="ui two buttons">
            <Button onClick={(e) => addDrink(name)} basic color="green">Add To Order</Button>
          </div>
        </Card.Content>
      </Card.Content>
    </Card>
  );
};

export default Drinks;
