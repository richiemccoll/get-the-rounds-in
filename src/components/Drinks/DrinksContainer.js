import React, { Component } from "react";
import { Header } from "semantic-ui-react";
import Drinks from "./Drinks";
import PropTypes from "prop-types";

class DrinksContainer extends Component {
  render() {
    const { drinks, prices, currentBar, addDrink } = this.props;

    const currentBarPrices = prices.filter(
      bar => bar.barId === currentBar[0].id
    );

    const currentBarDrinks = currentBarPrices.map(bar => {
      return drinks.map(drink => {
        if (drink.id === bar.productId) {
          const { name, image } = drink;
          const { price } = bar;
          return { name, image, price };
        }
      });
    });

    const availableDrinks = currentBarDrinks.map(i => i.filter(d => d));

    return (
      <div>
        <Header as="h2" textAlign="center">Available drinks</Header>
        <Drinks drinks={availableDrinks} addDrink={addDrink} />
      </div>
    );
  }
}

export default DrinksContainer;

DrinksContainer.PropTypes = {
  prices: PropTypes.array.isRequired,
  drinks: PropTypes.array.isRequired,
  addDrink: PropTypes.func.isRequired
};
