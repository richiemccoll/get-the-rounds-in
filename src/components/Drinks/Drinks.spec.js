import React from "react";
import Drinks from "./Drinks";
import DrinksContainer from "./DrinksContainer";
import renderer from "react-test-renderer";

describe("Drinks", () => {
  const selectedBar = {
    id: 1,
    image_url: "./potstill.jpg",
    name: "The Pot Still",
    address: "154 Hope Street",
    description: "A real local favourite, receiving more shout-outs from our followers than anywhere else. It’s a small pub with a massive drink selection, particularly if you’re in the mood for a whisky. ",
    lat: 0,
    long: 0,
    text: "The Pot Still",
    value: "0"
  };

  const bars = [
    {
      id: 1,
      image_url: "./potstill.jpg",
      name: "The Pot Still",
      address: "154 Hope Street",
      description: "A real local favourite, receiving more shout-outs from our followers than anywhere else. It’s a small pub with a massive drink selection, particularly if you’re in the mood for a whisky. ",
      lat: 0,
      long: 0,
      text: "The Pot Still",
      value: "0"
    },
    {
      id: 2,
      image_url: "./stereo.png",
      name: "Stereo & The Old Hairdressers",
      address: "Renfield Lane",
      description: "These are actually two separate bars, but they have the same vibe, same owner and are right next to each other. Both have a chilled out, stripped back aesthetic and well stocked bars.",
      lat: 0,
      long: 0,
      text: "Stereo & The Old Hairdressers",
      value: "1"
    }
  ];

  const drinks = [
    {
      id: 1,
      name: "Tennents Lager",
      image: "./tennents.jpg"
    },
    {
      id: 2,
      name: "Stella Artois",
      image: "./stella.jpeg"
    },
    {
      id: 3,
      name: "Jamesons Whiskey",
      image: "./jamesons.jpg"
    },
    {
      id: 4,
      name: "Jack Daniels & Coke",
      image: "./jackdaniels.jpg"
    },
    {
      id: 5,
      name: "Morgans Spiced & Lemonade",
      image: "./morgans.jpg"
    }
  ];

  const prices = [
    {
      barId: 1,
      productId: 2,
      price: 2.75
    },
    {
      barId: 1,
      productId: 5,
      price: 3.75
    }
  ];

  const addDrink = () => console.log("adding drink");

  it("renders the Drinks component and passes drinks, prices, currentBar and the addDrink function", () => {
    const component = renderer.create(
      <DrinksContainer
        currentBar={bars.filter(bar => bar.name === selectedBar.text)}
        drinks={drinks}
        prices={prices}
        addDrink={addDrink}
      />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
