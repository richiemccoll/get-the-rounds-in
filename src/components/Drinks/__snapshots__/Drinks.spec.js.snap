// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drinks renders the Drinks component and passes drinks, prices, currentBar and the addDrink function 1`] = `
<div>
  <h2
    className="ui header"
  >
    Available drinks
  </h2>
  <ul
    className="sc-bdVaJa gAOVPZ"
  >
    <li>
      <div
        className="ui card"
        href={undefined}
        onClick={[Function]}
      >
        <img
          alt={undefined}
          className="ui image"
          height={undefined}
          src="./stella.jpeg"
          width={undefined}
        />
        <div
          className="content"
        >
          <div
            className="header"
          >
            Stella Artois
          </div>
          <div
            className="description"
          >
            £
            2.75
          </div>
          <div
            className="extra content"
          >
            <div
              className="ui two buttons"
            >
              <button
                className="ui green basic button"
                onClick={[Function]}
                tabIndex={undefined}
              >
                Add To Order
              </button>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li>
      <div
        className="ui card"
        href={undefined}
        onClick={[Function]}
      >
        <img
          alt={undefined}
          className="ui image"
          height={undefined}
          src="./morgans.jpg"
          width={undefined}
        />
        <div
          className="content"
        >
          <div
            className="header"
          >
            Morgans Spiced & Lemonade
          </div>
          <div
            className="description"
          >
            £
            3.75
          </div>
          <div
            className="extra content"
          >
            <div
              className="ui two buttons"
            >
              <button
                className="ui green basic button"
                onClick={[Function]}
                tabIndex={undefined}
              >
                Add To Order
              </button>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
